src/Helpers/LeanCode.UserIdExtractors/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,89 @@
+using LeanCode.DomainModels.Ids;
+using LeanCode.UserIdExtractors.Extractors;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace LeanCode.UserIdExtractors;
+
+public static class UserServiceProviderExtensions
+{
+    public static IServiceCollection AddUserIdExtractor<TUserId>(
+        this IServiceCollection services,
+        string userIdClaim = "sub"
+    )
+        where TUserId : notnull, IEquatable<TUserId>
+    {
+        var userIdType = typeof(TUserId);
+        services.AddSingleton<IUserIdExtractor>(new StringUserIdExtractor(userIdClaim));
+
+        if (userIdType == typeof(string))
+        {
+            services.AddSingleton<IUserIdExtractor<string>>(new GenericStringUserIdExtractor(userIdClaim));
+        }
+        else if (userIdType == typeof(Guid))
+        {
+            services.AddSingleton<IUserIdExtractor<Guid>>(new GuidUserIdExtractor(userIdClaim));
+        }
+        else if (
+            !TryAddRawTypedUserIdExtractor(services, userIdType, userIdClaim)
+            && !TryAddPrefixedTypedUserIdExtractor(services, userIdType, userIdClaim)
+        )
+        {
+            throw new InvalidOperationException($"Type {userIdType} is not supported for user ID");
+        }
+
+        return services;
+    }
+
+    private static bool TryAddRawTypedUserIdExtractor(IServiceCollection services, Type userIdType, string userIdClaim)
+    {
+        var rawTypedId = GetImplementedIdTypes(userIdType, typeof(IRawTypedId<,>)).FirstOrDefault();
+
+        if (rawTypedId != null)
```

Comment:
```suggestion
        if (rawTypedId is not null)
```

I wonder if there's an analyzer for this somewhere :thinking: